### DISCUSSION TITLE
JASPAR annotation

### DISCUSSION THREAD

**Discussion by rojinsafavi** [GreenleafLab/ArchR#1344] [Questions / Documentation] [answered]
Hi @rcorces 
When I use JASPAR motif annotation, the addMotifAnnotations gives 633 TFs. I checked JASPAR database and there are around 1200 Homosapien TF. I was wondering if there is a specific reason that the number is much lower?

**Comment by rcorces** [comment, marked as answer]
ArchR uses bioconductor packages to obtain motif sets. For example, `JASPAR2020`.
https://github.com/GreenleafLab/ArchR/blob/968e4421ce7187a8ac7ea1cf6077412126876d5f/R/AnnotationPeaks.R#L330-L337

I'm not sure about the exact numbers but ArchR doesnt modify these motif sets. I'd recommend following up with the maintainers of those packages to understand any discrepancies.

```
> ArchR:::.requirePackage("JASPAR2020",installInfo='BiocManager::install("JASPAR2020")')
> args <- list(species = "Homo sapiens", collection = "CORE")
> motifs <- TFBSTools::getMatrixSet(JASPAR2020::JASPAR2020, args)
> motifs
PFMatrixList of length 633 names(633): MA0030.1 MA0031.1 MA0051.1 MA0057.1 MA0059.1 MA0066.1 MA0069.1 MA0070.1 ... MA0814.2 MA1123.2 MA0093.3 MA0526.3 MA0748.2 MA0528.2 MA0609.2

```

**Reply by rojinsafavi** [reply]
@rcorces  I looked at TFBTools, I think the reason the number is lower is that not all versions are used.

this piece of code will give 810 TF

```
args <- list(species = "Homo sapiens", collection = "CORE", all_versions = TRUE)
motifs <- TFBSTools::getMatrixSet(JASPAR2020::JASPAR2020, args)
PFMatrixList of length 810
names(810): MA0002.1 MA0003.1 MA0017.1 MA0018.1 ... MA0748.2 MA0528.2 MA0609.2

```
Is there a straightforward way to add this argument to ArchR?

**Reply by rcorces** [reply]
yes - if you look at the function documentation, you will see that we provide pass-through arguments to `TFBSTools::getMatrixSet`.

see `...` at the bottom of https://www.archrproject.com/reference/addMotifAnnotations.html

So you just add whatever additional params you want to send to `getMatrixSet()` into the params you pass to `addMotifAnnotations()`.

We allow pass-through arguments in most cases where ArchR calls another package's function (for ex. plotting functions that call ggPlot) but users seem to always miss this part of the documentation.

**Reply by rojinsafavi** [reply]
Thanks Ryan!

**Reply by LornaWessels** [reply]
Hi everyone, 
related to this I tried filtering the JASPAR2020 TF motif set by those included in the TF collection from Lambert et al. 2018 (to remove complexes etc.)

`The TFBSTools::getMatrixSet()`  function has the argument "name" which I can use to filter by the names of the TFs. 
However, this is not possible with `addMotifAnnotations()` since the argument "name" is already used by ArchR for the name of the stored annotation. 

I could still work around by passing a custom PWMatrixList though. 
But it would be super nice if this filtering option would work in `addMotifAnnotation()` directly. 

Best, 
Lorna

**Reply by rcorces** [reply]
@LornaWessels - I could make this change. Could you open up a new issue to track this?

**Comment by sylestiel** [comment]
@rcorces 


Can you help me figure out how to use the JASPAR2020 for addMotifAnnotations(). I tried the following and got an error.

args <- list(collection = "CORE", all_versions = TRUE)
motifL <- TFBSTools::getMatrixSet(JASPAR2020::JASPAR2020, args)
projC5.1 <- addMotifAnnotations(ArchRProj = projC5.1, motifSet ="JASPAR2020", name = "Motif",annoName = "motifL", force = TRUE)
[ArchR-addMotifAnnotations-ef322eebf59e-Date-2022-07-11_Time-17-41-20.log](https://github.com/GreenleafLab/ArchR/files/9087978/ArchR-addMotifAnnotations-ef322eebf59e-Date-2022-07-11_Time-17-41-20.log)

**Reply by rcorces** [reply]
@sylestiel - this is not the correct use of the function and its hard to know what you are trying to accomplish. See the function params.
https://github.com/GreenleafLab/ArchR/blob/f6c0388bd37023400794c9ae8562ad69e3ba9fd7/R/AnnotationPeaks.R#L273-L293
you do not need to call TFBSTools prior to addMotifAnnotations. Also, your log file does not include your error. The param `name` in your code is being passed to TFBSTools::getMatrixSet.

**Reply by sylestiel** [reply]
Thanks!
